views/BeerView.js

```diff
@@ -0,0 +1,67 @@
+var BeerView = Backbone.View.extend({
+
+  template: $('#beer-template'),
+
+  events: {
+    'click .remove-beer': 'delete',
+    'dblclick .name': 'toggleEditName',
+    'click .name-edit-submit' : 'editName',
+    'click .rate-submit': 'addRating'
+  },
+
+  initialize: function(){
+    this.listenTo(this.model, 'change:name', this.reName);
+    this.listenTo(this.model, 'change:ratings', this.reAvg);
+    this.listenTo(this.model, 'change:avgRate', this.renderAvg);
+  },
+
+  render: function(){
+    var template = Handlebars.compile(this.template.html());
+
+    this.$el.html(template(this.model.toJSON()));
+
+    return this;
+  },
+
+  delete: function(){
+    this.model.collection.remove(this.model);
+    this.remove();
+  },
+
+  toggleEditName: function(){
+    this.$el.find('.name-li').toggleClass('editing');
+    this.$el.find('.name-edit').find('input').focus();
+    this.$el.find('.name-edit').find('input').val('');
+  },
+
+  editName: function(e){
+    var newName = $(e.target).siblings('input').val();

```

Comment:
Again, you shouldn't have to use e.current target as the input is just a child of our `$el`. If you decide at some point to change your template html, this would break.